framework/src/restart/DataIO.C

```diff
@@ -416,3 +425,12 @@ dataLoad(std::istream & stream, std::stringstream * & s, void * context)
 {
   dataLoad(stream, *s, context);
 }
+
+template<>
+void
+dataLoad(std::istream & stream, MooseRandom & v, void * context)
+{
+  LIBMESH_BEST_UNORDERED_MAP<unsigned int, std::pair<mt_state, mt_state> > states;
+  loadHelper(stream, states, context);
```

Comment:
The common pattern is to make the dataLoad and dataStore methods _friends_ of the class. That way you can remove the getter and setter.